syntax = "proto3";
package proto_messages;

message SimpleMessage {
    int32 id = 1;
    repeated int32 array = 2;
    message OtherObj {
	    int32 id = 1;
    }
    OtherObj other_obj = 3;
}

message MessageRequest {
    SimpleMessage message = 1;
}

message MessageResponse {
    int32 id = 1;
    int32 size = 2;
    string type = 3;
}

service SimpleService {
    rpc HandleMessage(MessageRequest) returns (MessageResponse){};
}
